src/instructlab/model/dk_bench_utils.py

```
@@ -0,0 +1,310 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# pylint: disable=ungrouped-imports
+# Standard
+from datetime import datetime
+import enum
+import logging
+import os
+import pathlib
+
+# Third Party
+from instructlab.eval.mt_bench_common import get_openai_client
+from instructlab.eval.ragas import ModelConfig, RagasEvaluator
+from openai import OpenAI, OpenAIError
+from ragas.dataset_schema import EvaluationResult # type: ignore
+import click
+import httpx
+import pandas as pd
+
+# Local
+from ..client_utils import ClientException, http_client
+from ..configuration import DEFAULTS
+from ..utils import is_model_gguf, is_model_safetensors
+from .evaluate import get_model_name, launch_server
+
+logger = logging.getLogger(__name__)
+
+
+class IOFileType(enum.Enum):
+    CSV = "csv"
+    JSONL = "jsonl"
+    XLSX = "xlsx"
+    QNA_YAML = "qna.yaml"
+
+
+def model_is_localand_valid(model: str) -> bool:
+    valid_model = False
+    if model is not None and os.path.exists(model):
+        model_path = pathlib.Path(model)
+        if model_path.is_dir():
+            valid_model = is_model_safetensors(model_path)
+        elif model_path.is_file():
+            valid_model = is_model_gguf(model_path)
+    return valid_model
+
+
+def get_endpoint_model_name(endpoint: str, api_key: str | None, client: httpx.Client) -> str:
+    try:
+        openai_client = OpenAI(
+            base_url=endpoint,
+            api_key=api_key,
+            timeout=DEFAULTS.CONNECTION_TIMEOUT,
+            http_client=client,
+        )
+        models = openai_client.models.list()
+    except OpenAIError as exc:
+        raise ClientException(f"Connection Error {exc}") from exc
+
+    if len(models.data) != 1:
```

Why would we error if the server has more than one models actively running? What if I want to be evaluating multiple models concurrently?

---

My thinking here was that most OpenAI compatible endpoints, or at least in my research those served with vLLM only had one model served per endpoint.

I'd like it for the user to not have to think about passing in a model name at all and just get it for the server since getting the model name right is required to get the responses from the student model in the Eval library. I even had a flag for a model name that and then removed it for that reason. 

That being said in my next iteration of the PR I'm thinking of re-adding the --model-name flag and checking if that `model_name` is in the list of model names returned by the server. If it's not I'll throw an error. If no model_name is passed in and there are more than 1 models on the server I'll throw an error telling the user to pass in the model name from the ones returned by the server.

@RobotSail what do you think?

---

@alimaredia since you're already performing logic to check if the student model exists in the list of served models, you can either delete this check entirely _or_ have it only check that the server has at least one model being served.

In either of the proposed cases, you will have the same outcome of knowing whether or not a model exists. But it might help with UX if you can raise the error that the active server isn't hosting any models.